src/request.ts

```diff
@@ -0,0 +1,145 @@
+import {isObject} from 'lodash';
+import 'isomorphic-fetch';
+
+/**
+ * Return a promise that is resolved or rejected depending on the response's
+ * status code.
+ */
+export const checkStatus = (res: Response): Promise<Response> => {
+    if (res.status >= 300) {
+        return Promise.reject(res);
```

Comment:
I think it makes sense to return a Promise because the type definition calls for it. If we didn't i'd imagine you might get you'd get an error like checkStatus.then is not a function

Reply:
I would think so. This is how our current code works.

I'm thinking the reason an `Error` isn't thrown is in order to be able to pass the `res`. A vanilla `Error` only takes an error message so we wouldn't be able to pass the `res` and get out the info we need in later handlers.

I guess we could:

```js
throw res;
```

but we really should only be throwing error objects...

What do you think?

Reply:
@rwholey-eb `checkStatus` is used as a `.then` handler so anything anything thrown would just trigger a `.catch`. Technically I can just return `res` instead of returning `Promise.resolve(res)`

Reply:
ah i misunderstood, carry on

Reply:
Yeah it feels weird throwing an object. I was thinking the ideal would be returning Promise.reject(throw new Error(res.error)) but we're not guaranteed the response will contain an error and the current code may be using stuff from the raw response. Don't have a better solution yet :/